z80.h

```diff
@@ -450,6 +450,15 @@ class root {
     void on_tick(unsigned t) {
         unused(t); }
 
+    void on_reset_decoder() {}
+    void on_reset_cpu(bool soft = false) {
+        unused(soft);
+        self().on_reset_decoder(); }
+    void on_reset_memory() {}
+    void on_reset(bool soft = false) {
+        self().on_reset_cpu(soft);
+        self().on_reset_memory(); }
```

Comment:
@simonowen Shouldn't we leave memory as is on soft reset?

Reply:
Yes, soft reset doesn't affect RAM at all.  Initial state is more complicated and probably system-specific.

RAM is external to the CPU anyway, and emulators can choose to do something for hard reset if they wish.